paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -101,13 +101,16 @@ void ListenAndServOp::RunSyncLoop(
     framework::Scope *recv_scope,
     const std::vector<int> &prefetch_block_id_list) const {
   size_t num_blocks = program->Size();
+  auto skip_sub_blks = Attr<std::vector<int>>("skip_sub_blks");
```

Comment:
Record only the start block id and end block id for `listen_and_serv` op to run parallelly. Sub-blocks append to the end to make the program in a strict order.

Reply:
Another way is that we record the parallel block ids, the benefit is don't need `insert` block to the specified idx, just append block to pserver_program and record the optimize block id.

Reply:
Reuse that deprecated `optimize_block` attribute could be cool.

Reply:
Will do that.